tenacity/after.py

```diff
@@ -36,6 +36,8 @@ def after_log(
     """After call strategy that logs to some logger the finished attempt."""
 
     def log_it(retry_state: "RetryCallState") -> None:
+        if retry_state.fn is None:
+            return None
```

Comment:
This change breaks the existing tests (`TestAfterLogFormat.test_01_default` and `TestAfterLogFormat.test_02_custom_sec_format`).
In the real use case, `retry_state.fn` is not None (it should be the decorated function), but in tests the object is created by `make_retry_state` and the `fn` attribute is None by default.
Maybe we should manually add a dummy function to `fn` or we may use `Retrying` object to mock the decorator behavior.